app/rankers/algorithms.py

```diff
@@ -246,15 +248,41 @@ def user_impression_filter(recs: RecommendationListType, user_impressed_list: Li
 def rank_by_preferred_topics(
         recs: CorpusRecommendationListType,
         preferred_topics: List[TopicModel],
+        count: int
 ) -> CorpusRecommendationListType:
```

Comment:
**Optional:** It took me some time to convince myself that there is no possibility for an infinite loop, and that the function behaves as expected. Not required for this PR, but something to consider: could this function be composed of several logical components, that are more readable because they each perform a simple task? For example:
```python
rank_by_preferred_topics(
    spread_topics(
        randomize(recs)
    ),
    preferred_topics
)
```

1. `randomize` simply returns the input in random order.
2. `spread_topics` maximizes the space between recs of the same topic. It can be implemented very similar to what you've done, without the `random.randint` and `extend`, and by looping until the input is empty.
3. `preferred_topics` would behave the same as the original implementation of this function.